src/plugins/syscalls/syscalls.cpp

```diff
@@ -441,13 +510,45 @@ static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
         }
     }
 
-    print_header(s->format, drakvuf, info);
-    if ( nargs )
+    // store informations we need to pass to win_ret_cb()
+    wrapper->args = (unsigned char*)g_malloc(sizeof(char)*size);
+    memcpy(wrapper->args, buf, sizeof(char)*size);
+    wrapper->wsc = wsc;
+    wrapper->target_cr3 = info->regs->cr3;
+    if ( !drakvuf_get_current_thread_id(drakvuf, info->vcpu, &wrapper->target_tid) )
+        goto exit;
+
+    // read RSP to get the return address and add a trap on it
+    ctx.addr = info->regs->rsp;
+    if ( VMI_FAILURE == vmi_read(vmi, &ctx, s->reg_size, &ret_addr, NULL) )
+        goto exit;
+
+    PRINT_DEBUG("Adding trap to return address 0x%lx\n", s->ret_addr);
```

Comment:
Hi，Should we "s->ret_addr" ? Change to ret_addr .  PRINT_DEBUG("Adding trap to return address 0x%lx\n", ret_addr);

Reply:
Yeah you right